rskj-core/src/test/java/co/rsk/peg/union/UnionBridgeIT.java

```diff
@@ -0,0 +1,1101 @@
+        assertEquals(INITIAL_LOCKING_CAP, getUnionBridgeLockingCap());
+        // Assert that the stored union locking cap is still null since no increase has been made
+        assertNoStoredUnionLockingCap();
+        assertNoEventWasEmitted();
+    }
+
+    @Test
+    @Order(9)
+    void increaseUnionBridgeLockingCap_whenFirstVote_shouldVoteSuccessful() throws VMException {
+        // Arrange
+        setupCaller(CHANGE_LOCKING_CAP_AUTHORIZER_1);
+
+        // Act
+        int actualUnionResponseCode = increaseUnionBridgeLockingCap(NEW_LOCKING_CAP);
+
+        // Assert
+        assertEquals(UnionResponseCode.SUCCESS.getCode(), actualUnionResponseCode);
+        Coin actualUnionLockingCap = getUnionBridgeLockingCap();
+        assertEquals(INITIAL_LOCKING_CAP, actualUnionLockingCap);
+        assertNoStoredUnionLockingCap();
+        assertNoEventWasEmitted();
+    }
+
+    @Test
+    @Order(10)
+    void increaseUnionBridgeLockingCap_whenSecondVoteForDifferentValue_shouldVoteSuccessful() throws VMException {
+        // Arrange
+        setupCaller(CHANGE_LOCKING_CAP_AUTHORIZER_2);
+        // Act
+        int actualUnionResponseCode = increaseUnionBridgeLockingCap(DIFFERENT_LOCKING_CAP_FOR_VOTING);
+        // Assert
+        assertEquals(UnionResponseCode.SUCCESS.getCode(), actualUnionResponseCode);
+        Coin actualUnionLockingCap = getUnionBridgeLockingCap();
+        assertEquals(INITIAL_LOCKING_CAP, actualUnionLockingCap);
+        assertNoStoredUnionLockingCap();
+        assertNoEventWasEmitted();
+    }
+
+    @Test
+    @Order(11)
+    void increaseUnionBridgeLockingCap_whenThirdVoteForSameValue_shouldIncrementLockingCap() throws VMException {
+        // Act
+        int actualUnionResponseCode = increaseUnionBridgeLockingCap(NEW_LOCKING_CAP);
+        // Assert
+        assertEquals(UnionResponseCode.SUCCESS.getCode(), actualUnionResponseCode);
+        assertLockingCap(NEW_LOCKING_CAP);
+        assertLogUnionLockingCapIncreased(INITIAL_LOCKING_CAP, NEW_LOCKING_CAP);
+    }
+
+    @Test
+    @Order(12)
+    void increaseUnionBridgeLockingCap_whenVoteAgainForPreviousDifferentValueAfterElectionClear_shouldVoteSuccessfulButNoChange() throws VMException {
+        Coin lockingCapBeforeUpdate = getUnionBridgeLockingCap();
+        // Act
+        int actualUnionResponseCode = increaseUnionBridgeLockingCap(DIFFERENT_LOCKING_CAP_FOR_VOTING);
+        // Assert
+        assertEquals(UnionResponseCode.SUCCESS.getCode(), actualUnionResponseCode);
+        // Assert that the locking cap remains unchanged
+        assertLockingCap(lockingCapBeforeUpdate);
+        assertNoEventWasEmitted();
+    }
+
+    @Test
+    @Order(13)
+    void increaseUnionBridgeLockingCap_whenSmallerLockingCap_shouldReturnInvalidValue() throws VMException {
+        // Assert that the new locking cap is greater than the initial locking cap
+        assertEquals(-1, INITIAL_LOCKING_CAP.compareTo(NEW_LOCKING_CAP));
```

Comment:
```suggestion
        assertEquals(1, NEW_LOCKING_CAP.compareTo(INITIAL_LOCKING_CAP));
```
no big deal, just seems more aligned with the comment